### PR TITLE
1cc retarget

### PR DESCRIPTION
**Pull request by poojarazor**

### CODE REVIEW

File: includes/api/save-abandonment-data.php [hunk 1]
```diff
@@ -49,114 +49,360 @@ function saveCartAbandonmentData(WP_REST_Request $request)
...
+function saveWooAbandonmentCartLiteData($razorpayData, $wcOrderId)
+{
+    global $woocommerce;
+    global $wpdb;
+    $billingFirstName = $razorpayData['customer_details']['billing_address']['name'] ?? '';
+    $billingLastName = " ";
+    $billingZipcode = $razorpayData['customer_details']['billing_address']['zipcode'] ?? '';
+    $shippingZipcode  = $razorpayData['customer_details']['shipping_address']['zipcode'] ?? '';
+
+    $shippingCharges = WC()->cart->shipping_total;
```
Comment (ChetanGN) [inline includes/api/save-abandonment-data.php:103]: @poojarazor Where this cart object initiated? Will it calculate the shipping charges properly?

File: includes/api/save-abandonment-data.php [hunk 2]
```diff
@@ -49,114 +49,360 @@ function saveCartAbandonmentData(WP_REST_Request $request)
...
-                }
+        $cartInfo = wp_json_encode($cart);
+        $results =  checkRecordBySession($getCookie[0]);
+
+        if (get_post_meta($wcOrderId, 'abandoned_user_id', true) == '')
+        {
+            add_post_meta( $wcOrderId, 'abandoned_user_id', $userId); }
+        else
+        {
+            update_post_meta( $wcOrderId, 'abandoned_user_id', $userId );
```
Comment (ChetanGN) [inline includes/api/save-abandonment-data.php:130]: @poojarazor I think `update_post_meta` will check the key existence, so please confirm whether `add_post_meta` is required or not.

File: includes/api/save-abandonment-data.php [hunk 3]
```diff
@@ -49,114 +49,360 @@ function saveCartAbandonmentData(WP_REST_Request $request)
...
-                    $statusCode          = 400;
-                } else {
-                    // Storing session_id in WooCommerce session.
-                    WC()->session->set('wcf_session_id', $sessionId);
-                    $response['status']  = true;
-                    $response['message'] = 'Data successfully inserted for wooCommerce cart abandonment recovery';
-                    $statusCode          = 200;
-                }
+        $cartInfo = wp_json_encode($cart);
+        $results =  checkRecordBySession($getCookie[0]);
```
Comment (ChetanGN) [inline includes/api/save-abandonment-data.php:123]: Why same var names has been used here?

File: includes/api/save-abandonment-data.php (hunk 1 above)
Comment (ChetanGN) [reply includes/api/save-abandonment-data.php]: @poojarazor Update on this?

File: woo-razorpay.php [hunk 4]
```diff
@@ -1846,7 +1851,7 @@ function razorpay_webhook_init()
 }
 
 define('RZP_PATH', plugin_dir_path( __FILE__ ));
-define('RZP_CHECKOUTJS_URL', 'https://checkout.razorpay.com/v1/checkout.js');
+define('RZP_CHECKOUTJS_URL', 'https://betacdn.np.razorpay.in/checkout/builds/branch-builds/1cc/kinsta-test-wc/v1/checkout.js');
```
Comment (ChetanGN) [inline woo-razorpay.php:1854]: @poojarazor Please don't do this kind of mistake. Update it.

File: woo-razorpay.php [hunk 5]
```diff
@@ -1666,21 +1666,25 @@ protected function updateRecoverCartInfo($wcOrderId)
             if (isset($cut_off_time))
             {
                 $cartCutOffTime = intval($cutOffTime) * 60;
+                add_post_meta(363, 'test', 'yes');
```
Comment (ChetanGN) [inline woo-razorpay.php:1669]: What is this?

File: woo-razorpay.php (hunk 4 above)
Comment (poojarazor) [reply woo-razorpay.php]: got it

File: woo-razorpay.php (hunk 5 above)
Comment (poojarazor) [reply woo-razorpay.php]: removed

**Review by ChetanGN — APPROVED** [review]
LGTM

**Pull request merged**